shared/hash.c

```diff
@@ -249,6 +249,8 @@ int hash_del(struct hash *hash, const char *key)
 		.value = NULL,
 	};
 
+	if (bucket->entries == NULL)
+		return -ENOENT;
```

Comment:
Nit: please add a blank line.

Similar to the previous commit, can we get a test. Massive thanks in advance.